bundles/org.openhab.misc.addonsuggestionfinder/src/main/java/org/openhab/misc/addonsuggestionfinder/AddonSuggestionInfoProvider.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.misc.addonsuggestionfinder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.addon.AddonInfoList;
+import org.openhab.core.addon.AddonInfoListReader;
+import org.openhab.core.addon.AddonInfoProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AddonSuggestionInfoProvider} provides a list of candidate suggested addons to be installed.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(name = "addon-suggestion-info-provider", service = AddonInfoProvider.class)
+public class AddonSuggestionInfoProvider implements AddonInfoProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(AddonSuggestionInfoProvider.class);
+    private final Set<AddonInfo> candidateAddonInfos = new HashSet<>();
+
+    @Activate
+    public AddonSuggestionInfoProvider() {
+        setCandidates(getResourceXml());
+    }
+
+    @Override
+    public @Nullable AddonInfo getAddonInfo(@Nullable String id, @Nullable Locale locale) {
+        return candidateAddonInfos.stream().filter(a -> a.getId().equals(id)).findFirst().orElse(null);
```

Comment:
Shouldn't this be `getUID()`?
When comparing to the `XmlAddonInfoProvider` it ultimately matches on uid, not id.

Reply:
Confirmed, id in AddonInfo should match uid in Addon. You therefore do not need a separate uid field in the suggestions, but the id field should be the uid from Addon.

Reply:
Yeah. I was confused by the argument name in the java doc, so I interpreted it wysiwyg, even though it didn't make real sense. I will change it.

Reply:
Done